key.ui/src/main/java/de/uka/ilkd/key/core/KeYMediator.java

```diff
@@ -304,26 +211,17 @@ public String toString() {
     /**
      * Selects the specified proof and initializes it.
      *
-     * @param p the proof to select.
+     * @param newProof the proof to select.
+     * @param previousProof the previously selected proof
      */
-    public void setProof(Proof p) {
-        if (getSelectedProof() == p) {
+    private void setProof(Proof newProof, Proof previousProof) {
```

Comment:
What is the purpose of this additional `previousProof` parameter? Concurrent `setProof` calls?